src/components/drawing/index.js

```diff
@@ -354,6 +354,139 @@ drawing.gradient = function(sel, gd, gradientID, type, colorscale, prop) {
     fullLayout._gradientUrlQueryParts[k] = 1;
 };
 
+/**
+ * pattern: create and apply a pattern fill
+ *
+ * @param {object} sel: d3 selection to apply this pattern to
+ *     You can use `selection.call(Drawing.pattern, ...)`
+ * @param {DOM element} gd: the graph div `sel` is part of
+ * @param {string} patternID: a unique (within this plot) identifier
+ *     for this pattern, so that we don't create unnecessary definitions
+ * @param {string} bgcolor: background color for this pattern
+ * @param {string} fgcolor: foreground color for this pattern
+ * @param {number} scale: scale of this pattern
+ * @param {number} solidity: how solid lines of this pattern are
+ * @param {string} prop: the property to apply to, 'fill' or 'stroke'
+ */
+drawing.pattern = function(sel, gd, patternID, shape, bgcolor, fgcolor, scale, solidity, prop) {
+    var fullLayout = gd._fullLayout;
+    var fullID = 'p' + fullLayout._uid + '-' + patternID;
+    var baseSize = 8 * scale;
+    var width, height;
+
+    var path = '';
+    switch(shape) {
+        case '/':
+            width = baseSize * Math.sqrt(2);
+            height = baseSize * Math.sqrt(2);
+            path = 'M-1,1l2,-2' +
+                   'M0,' + height + 'L' + width + ',0' +
+                   'M' + (width - 1) + ',' + (height + 1) + 'l2,-2';
```

Comment:
This path (and the other diagonal patterns) needs tweaking for very high solidity or it'll lose some fill at the corners - simplest is probably to extend the little corner lines eg `'M-' + (width / 4) + ',' + (height / 4) + 'l' + (width / 2) + ',-' + (height / 2)`

Reply:
Absolutely. Fixed in 39e35ccc02eb5d76ea7bbdc88be61a7cb1943918.